.github/workflows/reusable_testground_test.yaml

```diff
@@ -0,0 +1,113 @@
+name: TestGround Tests
+
+on:
+  workflow_call:
+    inputs:
+      tests_files_path:
+        description: "Path to Tests"
+        required: true
+        type: string
+        default: "compositions/cluster-k8s/sanity/002-da-sync-8.toml"
+      testground_url:
+        required: true
+        type: string
+        description: "TestGround LoadBalancer URL"
+      testground_region:
+        required: true
+        type: string
+        description: "TestGround Cluster Region"
+        default: "eu-west-1"
+      celestia_node_commit_sha:
+        required: false
+        type: string
+        description: "You can specify any commit that you want to use in the celestia-node repository"
+
+env:
+  REPO_TG: https://github.com/testground/testground.git
+  REPO_TEST_TG: https://github.com/celestiaorg/test-infra.git
+  ORG: celestiaorg
+  CELESTIA_NODE_REPO: celestia-node
+  BRANCH: main
+  TG_USER: ga-bot
+  TESTGROUND_URL: ${{ inputs.testground_url }}
+  TESTGROUND_REGION: ${{ inputs.testground_region }}
+  TEST_FILES_PATH: ${{ inputs.tests_files_path }}
+ 
+jobs:
+  testground:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Clone the repo
+        run: |
+          git clone ${{ env.REPO_TG }}
+          git clone ${{ env.REPO_TEST_TG }}
+
+      - name: Build TestGround
+        run: |
+          export GOBIN=$(go env GOPATH)/bin
+          mkdir -p $GOBIN
+          cd testground
+          make goinstall
+
+      - name: Generate Config TestGround
+        run: |
+          mkdir -p ~/.config/testground/
+          cat <<EOF > ~/.config/testground/.env.toml
+          [aws]
+          region="${{ env.TESTGROUND_REGION }}"
+          [client]
+          endpoint = "${{ env.TESTGROUND_URL }}"
+          user="${{ env.TG_USER }}"
+          EOF
+          cat ~/.config/testground/.env.toml
+
+      # In the next step, we can verify if we need to use an specific commit 
+      # or we can use the one in the file.
+      - name: Update go.mod with the latest Celestia Node commit
```

Comment:
So this only supports node right now?
- What happens if one wants to test a specific combination between app & node?
- What if app wants to use this workflow and they don't need node?

Reply:
I did it only to modify the node, but I can also add the app following the same approach

- What happens if one wants to test a specific combination between app & node?
  - In that case, we can only modify the node, and it will use the app version which is defined in the [go.mod](https://github.com/celestiaorg/test-infra/blob/main/go.mod#L15)
- What if app wants to use this workflow and they don't need node?
  - if I'm not wrong, this will be managed in the tests, we have the `go.mod` for the project, but you can specify the tests to run only the celestia-app services